pkg/container-hook/tracer.go

```diff
@@ -447,40 +473,36 @@ func (n *ContainerNotifier) watchPidFileIterate(
 		return false, err
 	}
 
-	// cri-o appends userdata to bundleDir,
-	// so we trim it here to get the correct containerID
-	containerID := filepath.Base(filepath.Clean(strings.TrimSuffix(bundleDir, "userdata")))
-
-	err = n.AddWatchContainerTermination(containerID, containerPID)
+	err = n.AddWatchContainerTermination(pc.id, containerPID)
 	if err != nil {
-		log.Errorf("container %s with pid %d terminated before we could watch it: %s", containerID, containerPID, err)
-		return true, nil
+		log.Errorf("container %s with pid %d terminated before we could watch it: %s", pc.id, containerPID, err)
+		return false, nil
 	}
 
 	if containerPID > math.MaxUint32 {
 		log.Errorf("Container PID (%d) exceeds math.MaxUint32 (%d)", containerPID, math.MaxUint32)
-		return true, nil
+		return false, nil
 	}
 
 	var containerName string
 	n.futureMu.Lock()
-	fc, ok := n.futureContainers[containerID]
+	fc, ok := n.futureContainers[pc.id]
 	if ok {
 		containerName = fc.name
 	}
-	delete(n.futureContainers, containerID)
+	delete(n.futureContainers, pc.id)
 	n.futureMu.Unlock()
 
 	n.callback(ContainerEvent{
 		Type:            EventTypeAddContainer,
-		ContainerID:     containerID,
+		ContainerID:     pc.id,
 		ContainerPID:    uint32(containerPID),
 		ContainerConfig: containerConfig,
-		Bundle:          bundleDir,
+		Bundle:          pc.bundleDir,
 		ContainerName:   containerName,
 	})
 
-	return true, nil
+	return false, nil
```

Comment:
Ok, I will document it.

The processing should stop if it is an error that prevent processing of further fanotify events on unrelated containers.

If it is an error on the current container only (e.g. cannot parse its config.json or it was deleted under our feet), we should continue because the fanotify fd is used for subsequent containers.